### ISSUE TITLE
Why background class is not used for semantic segmentation loss?

### ISSUE THREAD

**Issue by alexbuyval** [dbolya/yolact#75]
Hi

I am trying to use the YOLACT for simultaneous object detection and road detection. According to the @dbolya  answer [here](https://github.com/dbolya/yolact/issues/73), I am trying to use hidden segmentation head.
However, I have found that the output segmentation layer produces cfg.num_classes-1 features.
[`self.semantic_seg_conv = nn.Conv2d(src_channels[0], cfg.num_classes-1, kernel_size=1)`](https://github.com/dbolya/yolact/blob/master/yolact.py#L456)
 So it means that there is no background in the semantic mask.

In my current dataset I have the following classes: ('road', 'person', 'rider', 'car', 'truck', 'bus', 'train', 'motorcycle', 'bicycle'), so there are unlabeled areas like building, sky and so on.

@dbolya Could you explain to me how to manage this problem? 
I see two approaches:
1. Add to the dataset sky, building to fill all unlabeled areas. But it difficult to label.
2. Add one feature to output segmentation layer. However, I am not sure how to create a target mask in this case [here](https://github.com/dbolya/yolact/blob/master/data/coco.py#L140).

Thank you in advance,
Alex

**Comment by dbolya** [comment]
Are you sure you want to explicitly train a background class? Wouldn't it be better to do something like 
```python
bg = 1 - (pred_outs['segm'] > 0).float().max(dim=1)[0]
```
(i.e., one minus the union of the foreground).

**Comment by alexbuyval** [comment]
@dbolya  Thank you for your fast answer

In fact, I don't need the background class at all :)

I just afraid that it will impact the learning of the segmentation mask. So if it is ok, I will leave it as is.
